openstack/barbican/templates/barbican-nanny-deployment.yaml

```diff
@@ -0,0 +1,112 @@
+{{- if .Values.barbican_nanny.enabled }}
+{{- if .Values.barbican_nanny.db_secret_move.enabled }}
+kind: Deployment
+apiVersion: apps/v1
+metadata:
+  name: barbican-nanny
+  labels:
+    app: {{ template "name" . }}
+    chart: "{{ .Chart.Name }}-{{ .Chart.Version }}"
+    release: "{{ .Release.Name }}"
+    heritage: "{{ .Release.Service }}"
+    type: nanny
+    component: barbican
+spec:
+  replicas: {{ .Values.barbican_nanny.replicas }}
+  revisionHistoryLimit: {{ .Values.pod.lifecycle.upgrades.deployments.revisionHistory }}
+  strategy:
+    type: {{ .Values.pod.lifecycle.upgrades.deployments.podReplacementStrategy }}
+    {{ if eq .Values.pod.lifecycle.upgrades.deployments.podReplacementStrategy "RollingUpdate" }}
+    rollingUpdate:
+      maxUnavailable: {{ .Values.pod.lifecycle.upgrades.deployments.rollingUpdate.maxUnavailable }}
+      maxSurge: {{ .Values.pod.lifecycle.upgrades.deployments.rollingUpdate.maxSurge }}
+    {{ end }}
+  selector:
+    matchLabels:
+      app: barbican-nanny
+  template:
+    metadata:
+      labels:
+        app: barbican-nanny
+        name: barbican
+        alert-tier: os
+        alert-service: barbican
+        {{- include "utils.topology.pod_label" . | indent 8 }}
+      annotations:
+        chart-version: {{.Chart.Version}}
+        checksum/etc-configmap.conf: {{ include (print $.Template.BasePath "/etc-configmap.yaml") . | sha256sum }}
+        secrets-hash: {{ include (print $.Template.BasePath "/secrets.yaml") . | sha256sum }}
+        {{- if .Values.proxysql.mode }}
+        prometheus.io/scrape: "true"
+        prometheus.io/targets: {{ required ".Values.alerts.prometheus missing" .Values.alerts.prometheus | quote }}
+        {{- end }}
+        {{- include "utils.linkerd.pod_and_service_annotation" . | indent 8 }}
+    spec:
+      {{ include "utils.proxysql.pod_settings" . | indent 6 }}
+{{- if .Values.barbican_nanny.db_secret_move.enabled }}
```

Comment:
Does that `if` make sense here? It would remove all the `containers` currently. I propose to remove it and only add inidividual `if`s once you have more than one container in this nanny. Then, you'll probably also want to move the `interval` config into the individual nannies.

Reply:
we have plan to add vcenter-barbican nanny hence this was introduced.

Reply:
Ok, but then this `if` needs to go below `containers:` as either vcenter or move-secrets or both can be enabled and enabling both would otherwise duplicate the `containers` key or enabling one would be missing the `containers` key.